File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if (!head || !head->next) return head;

        ListNode *p = NULL;
        ListNode *temp = head;
        ListNode *n = temp->next;

        while(n){
            temp->next = p;
            p = temp;
            temp = n;
            n = n->next;
        }
        temp->next = p;

        return temp;

    }
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head
        
        p = None
        temp = head
        n = temp.next

        while(n):
            temp.next = p
            p = temp
            temp = n
            n = n.next
        
        temp.next = p

        return temp